istio-telemetry/mixer-telemetry/templates/service.yaml

```diff
@@ -18,7 +18,8 @@ spec:
   - name: prometheus
     port: 42422
   selector:
-    app: istio-telemetry
+    istio: mixer
```

Comment:
after spending some time with debugging why mixer is not getting any metrics, I got into the same conclusion as the service/pod labels don't match. While I was planning to push the change, I noticed this PR already exists. 

At least, I can confirm that the label changes you have here works as expected. 👍